llvm/lib/Support/ErrorHandling.cpp

```diff
@@ -39,12 +40,37 @@
 
 using namespace llvm;
 
+class MemoryReserver {
+private:
+  std::error_code EC;
+  sys::MemoryBlock Ptr;
+
+public:
+  MemoryReserver(size_t Pages): Ptr() {
+    using sys::Memory;
+    unsigned int PSize = sys::Process::getPageSizeEstimate();
+    Ptr = Memory::allocateMappedMemory(PSize * Pages, nullptr, Memory::MF_READ | Memory::MF_WRITE, EC);
+    assert(!EC);
+  }
+
+  ~MemoryReserver() {
+    release();
+  }
+
+  void release() {
+    EC = sys::Memory::releaseMappedMemory(Ptr);
+    assert(!EC);
```

Comment:
In case `.release` is explicitly called it will be called again upon destruction. Or maybe it could just be called twice for $REASON.

Guard the release with a `if (Ptr != nullptr)` and reset `Ptr`.

Reply:
It's already handled by `releaseMappedMemory`
https://github.com/revng/llvm-project/blob/master/llvm/lib/Support/Unix/Memory.inc#L152